base/essentials.jl

```diff
@@ -151,9 +151,29 @@ macro isdefined(s::Symbol)
     return Expr(:escape, Expr(:isdefined, s))
 end
 
+# can be used in place of `@pure` (supposed to be used for bootstrapping)
 macro _pure_meta()
     return Expr(:meta, :pure)
 end
+# can be used in place of `@assume_effects :total` (supposed to be used for bootstrapping)
+macro _total_meta()
+    return Expr(:meta, Expr(:purity,
```

Comment:
```suggestion
    __module__ === Base && return Expr(:meta, Expr(:purity,
```
:trollface:

Reply:
Aha, seems to be a good idea. Or more precisely
```julia
function _is_internal(__module__)
    if ccall(:jl_base_relative_to, Any, (Any,), __module__)::Module === Core.Compiler ||
       nameof(__module__) === :Base
        return true
    end
    return false
end
```
?